Fix off-by-one in RetryingChannel, use the configured number of retries

carterkozak opened the pull request:
==COMMIT_MSG==
Fix off-by-one in RetryingChannel, use the configured number of retries
==COMMIT_MSG==

## Possible downsides?
Doesn't support the configured backoff interval yet.

carterkozak committed:
Add generated changelog entries

carterkozak committed:
Fix off-by-one in RetryingChannel, use the configured number of retries

carterkozak commented on dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java:
```diff
@@ -59,23 +48,21 @@
     }
 
     private final class RetryingCallback implements FutureCallback<Response> {
         private final AtomicInteger failures = new AtomicInteger(0);
```
This should be an `int` given that retries are sequenced we can rely on the happens-before relationship.

ferozco replied:
do you want to make that change as part of this PR?

carterkozak replied:
I'll add it to a an improvement on top of this PR, might be best if you'd like to review that in isolation :-]

ferozco commented:
👍

carterkozak committed:
fix #311: RetryingChannel propagates cancelation (#368)

* fix #311: RetryingChannel propagates cancelation

Generally safer by using existing utility functionality to chain
futures instead of reimplementing async-transformation via
callbacks.
